v3/src/components/data-display/components/legend/categorical-legend.tsx

```diff
@@ -64,8 +65,8 @@ export const CategoricalLegend =
       if (caseIds) {
         // This is breaking the graph-legend cypress test
         // setOrExtendSelection(caseIds, dataConfiguration?.dataset, event.shiftKey)
-        if (event.shiftKey) dataConfiguration?.dataset?.selectCases(caseIds)
-        else dataConfiguration?.dataset?.setSelectedCases(caseIds)
+        if (event.shiftKey) setSelectedCases(caseIds, dataConfiguration?.dataset)
```

Comment:
```suggestion
        if (event.shiftKey) selectedCases(caseIds, dataConfiguration?.dataset)
```
Otherwise, you'll set the selection rather than extending it.